Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp

```diff
@@ -484,6 +524,70 @@ namespace AzToolsFramework::Prefab
         }
     }
 
+    void PrefabFocusHandler::SetInstanceContainersOpenStateOfAllDescendantContainers(
+        InstanceOptionalReference instance, bool openState) const
+    {
+        // If this is called outside the Editor, this interface won't be initialized.
+        if (!m_containerEntityInterface)
+        {
+            return;
+        }
+
+        if (!instance.has_value())
+        {
+            return;
+        }
+
+        AZStd::queue<InstanceOptionalReference> instanceQueue;
+
+        // We skip this instance and start from children.
+        instance->get().GetNestedInstances(
+            [&](AZStd::unique_ptr<Instance>& nestedInstance)
+        {
+            instanceQueue.push(*nestedInstance.get());
+        }
+        );
+
+        while (!instanceQueue.empty())
+        {
+            InstanceOptionalReference currentInstance = instanceQueue.front();
+            instanceQueue.pop();
+
+            if (currentInstance.has_value())
+            {
+                m_containerEntityInterface->SetContainerOpen(currentInstance->get().GetContainerEntityId(), openState);
+
+                currentInstance->get().GetNestedInstances(
+                    [&](AZStd::unique_ptr<Instance>& nestedInstance)
+                {
+                    instanceQueue.push(*nestedInstance.get());
+                }
+                );
+            }
+        }
+    }
+
+    void PrefabFocusHandler::SwitchToEditScope() const
+    {
+        auto focusInstance = GetInstanceReference(m_rootAliasFocusPath);
+
+        switch (m_prefabEditScope)
+        {
+        case PrefabEditScope::HIDE_NESTED_INSTANCES_CONTENT:
+        {
+            SetInstanceContainersOpenStateOfAllDescendantContainers(focusInstance, false);
+        }
+        break;
+        case PrefabEditScope::SHOW_NESTED_INSTANCES_CONTENT:
+        {
+            SetInstanceContainersOpenStateOfAllDescendantContainers(focusInstance, true);
+        }
+        break;
+        }
+
```

Comment:
nit: add a `default` case for `HIDE_NESTED_INSTANCES_CONTENT`

Reply:
Updated.